lnwallet/wallet.go

```diff
@@ -579,8 +579,16 @@ func (l *LightningWallet) PsbtFundingVerify(pendingChanID [32]byte,
 	// but only if we are contributing funds to the channel. This is done
 	// to still allow incoming channels even though we have no UTXOs
 	// available, as in bootstrapping phases.
+	//
+	// We also increment the counter when _accepting_ an incoming anchor
+	// channel and the node is configured for HTLC forwarding. This ensures
+	// that we attempt to maintain an onchain UTXO for fee bumping in the
+	// event of unilateral closure. Not maintain a fee bumping UTXO is
+	// dangerous when forwarding payments, since the node may not be able to
+	// sweep a disputed HTLC on the upstream channel with the preimage,
+	// resulting in loss of funds.
 	if pendingReservation.partialState.ChanType.HasAnchors() &&
-		intent.LocalFundingAmt() > 0 {
+		(intent.LocalFundingAmt() > 0 || l.Cfg.IsRoutingNode) {
```

Comment:
this check must also be done in the wallet-funding scenario elsewhere in this file

Reply:
good catch, i've added a integration test to test the basic funding flow